tests/unit/web/ng/impl/WsConnectionTests.cpp

```diff
@@ -108,32 +110,38 @@ TEST_F(WebWsConnectionTests, WasUpgraded)
     });
 }
 
-// This test is either flaky or incorrect
-// see https://github.com/XRPLF/clio/issues/2700
-TEST_F(WebWsConnectionTests, DISABLED_DisconnectClientOnInactivity)
+TEST_F(WebWsConnectionTests, DisconnectClientOnInactivity)
 {
     boost::asio::io_context clientCtx;
     auto work = boost::asio::make_work_guard(clientCtx);
     std::thread clientThread{[&clientCtx]() { clientCtx.run(); }};
 
-    util::spawn(clientCtx, [&work, this](boost::asio::yield_context yield) {
+    std::mutex mutex;
+    std::condition_variable cv;
+    bool finished{false};
+
+    util::spawn(clientCtx, [&](boost::asio::yield_context yield) {
         auto expectedSuccess =
             wsClient_.connect("localhost", httpServer_.port(), yield, std::chrono::milliseconds{100});
         [&]() { ASSERT_TRUE(expectedSuccess.has_value()) << expectedSuccess.error().message(); }();
-        boost::asio::steady_timer timer{yield.get_executor(), std::chrono::milliseconds{5}};
-        timer.async_wait(yield);
+        std::unique_lock lock{mutex};
+        auto const gotNotified = cv.wait_for(lock, std::chrono::seconds{2}, [&finished]() { return finished; });
```

Comment:
Just `wait` will also work if the test succeeds.
But I do understand the desire to limit the test execution.

Let's maybe add a comment why we use time limit here?